lib/rules/rule-nesting-at-rule-required-list/index.mjs

```diff
@@ -0,0 +1,80 @@
+import { isAtRule, isRoot } from '../../utils/typeGuards.mjs';
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import isStandardSyntaxRule from '../../utils/isStandardSyntaxRule.mjs';
+import matchesStringOrRegExp from '../../utils/matchesStringOrRegExp.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'rule-nesting-at-rule-required-list';
+
+const messages = ruleMessages(ruleName, {
+	expected: (atRulePatterns) =>
+		`Expected rule to be nested in an at-rule matching pattern(s) "${atRulePatterns}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/rule-nesting-at-rule-required-list',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary) => {
+	return (root, result) => {
+		// Validate that the primary option is a string, regex, or array of strings/regexes
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: [isString, isRegExp],
+		});
+
+		if (!validOptions) return;
+
+		root.walkRules((ruleNode) => {
+			if (!isStandardSyntaxRule(ruleNode)) return;
+
+			// Skip if rule already has a required at-rule ancestor
+			if (hasRequiredAtRuleAncestor(ruleNode, primary)) return;
+
+			report({
+				message: messages.expected,
+				messageArgs: [Array.isArray(primary) ? primary.join(', ') : primary],
```

Comment:
question

@ybiquitous with the removal of [`util.format`](https://nodejs.org/api/util.html#utilformatformat-args) in #6389 we don't support `%o` anymore.
Do we pass a string instead or add back `util.format`?